code/modules/admin/topic.dm

```diff
@@ -1691,3 +1691,9 @@ mob/living/silicon/ai/can_centcom_reply()
 
 	. = "<A HREF='?[source];adminplayerobservejump=\ref[target]'>JMP</A>"
 	. += target.extra_admin_link(source)
+
+/proc/spawn_humanoid_species_admin(var/mob/M, var/delmob)

```

Comment:
ohnonono. This proc, as written, is a security risk.

First, non-interface procs should not use `usr`. The user should be a a mob or a client. In this case probably a mob. And the mob's permissions should be validated. since href hacking is a thing. Note that input's first (hidden) argument is a mob, so you can easily do `input(some_user, "bla")`

Reply:
What do you mean usr? This proc doesn't use usr, it is being passed as argument from main proc above. And that proc gets it from the topic from topic.dm and href hack is a security risk right now as well.

Reply:
Done